app/src/main/java/com/example/drawingactivity/DrawActivity.kt

```diff
@@ -249,4 +303,73 @@ class DrawActivity : AppCompatActivity() {
             }
         }
     }
+
+    /**
+     * This function is used to setup all the behavior for the bottomSheet.
+     * It sets all of the views in the bottom sheet to set a layout to be visible and expands the sheet.
+     * after making the layout visible, it will animate in via a fade in.
+     * If any button is clicked afterwards, all layouts have animations cancelled.
+     */
+    private fun setupBottomSheet(
+        context: Context,
+        bottomSheet: LinearLayout,
+        viewList: List<View>,
+        layoutList: List<View>
+    ) {
+        behavior = BottomSheetBehavior.from(bottomSheet)
+        viewList.forEach {
```

Comment:
Why have them in a list if the list is just going to put a giant when block into each click listener. This is less efficient than just setting each of them manually lol

Consider, you keep getting layoutList[0], layoutList[1] etc below. That makes it very inflexible to changing the order later
You could just use forEachIndexed to get the index of what was clicked, or indexOf the list. There's no need to hardcode list indices 

Also I'd rather see a better mapping of the enum to the View. Why not make a data class that holds a View and your enum and iterate over that rather than doing awkward redundant mapping here?

`data class ToolButtonData(view: View, toolType: Whateveryourenumiscalled)`

This works here, but it's not elegant. Let's give it another go

Reply:
talked in class, resolved this